packages/snap/src/types.ts

```diff
@@ -0,0 +1,52 @@
+import { Bytes } from '@metamask/utils';
+
+/**
+ * The parameters for calling the `getPublicKey` JSON-RPC method.
+ *
+ * Note: For simplicity, these are not validated by the snap. In production, you
+ * should validate that the request object matches this type before using it.
```

Comment:
As noted by the comment, type validation is required here. It's very much okay to track this in a separate issue without further modifications to the current PR, but I want to stress how important it is for the Snap to validate every last piece of data it receives from RPC. It's very easy to introduce security holes if we don't.

Upstream Metamask Snaps sources seem to rely on `superstruct` quite heavily, and they actually provide some helpers here. Our validation logic may look something like this:

```typescript
import { Bytes } from '@metamask/utils';
import { Bip32PathStruct } from '@metamask/snaps-utils/*';
import { Infer, boolean, enums, object, optional, type } from 'superstruct';

export const SupportedCurveEnums = enums(['ed25519', 'sec256k1']);

/**
 * The supported curves for deriving a BIP-32 account.
 */
export type SupportedCurve = Infer<typeof SupportedCurveEnums>;

/**
 * `type` is used instead of `object` to allow unknown properties.
 */
export const GetBip32PublicKeyParamsStruct = type({
  /**
   * The BIP-32 path to the account.
   */
  path: Bip32PathStruct,
  /**
   * The curve used to derive the account.
   */
  curve: SupportedCurveEnums,
  /**
   * Whether to return the public key in compressed form.
   */
  compressed: optional(boolean()),
});

/**
 * The parameters for calling the `getPublicKey` JSON-RPC method.
 *
 * Unknown properties are ignored and passed to `snap_getBip32PublicKey`.
 */
export type GetBip32PublicKeyParams = Infer<
  typeof GetBip32PublicKeyParamsStruct
>;

export const SignMessageParamsStruct = object({
  /**
   * The BIP-32 path to the account.
   */
  message: Bytes,
  /**
   * The curve used to derive the account.
   */
  curve: SupportedCurveEnums,
  /**
   * The BIP-32 path to the account.
   */
  path: Bip32PathStruct,
});

/**
 * The parameters for calling the `signMessage` JSON-RPC method.
 */
export type SignMessageParams = Infer<typeof SignMessageParamsStruct>;
```

Reply:
Sounds good, leaving an open issue to tackle this!

Reply:
Perfect, I think we're good! Approving.

Reply:
https://github.com/Sovereign-Labs/sov-snap/issues/2